src/opnsense/mvc/app/views/OPNsense/Unbound/overview.volt

```diff
@@ -553,83 +554,143 @@
+                dialogRef.setMessage($container);
+
+                const $body = dialogRef.getModalBody();
+                $body.off('click', '.policy-action').on('click', '.policy-action', function () {
+                    const $b = $(this);
+                    $('.tooltip:visible').hide();
+
+                    ajaxCall('/api/unbound/settings/update_blocklist', {
+                        domain: $b.data('domain'),
+                        uuid: $b.data('uuid'),
+                        type: $b.data('type')
+                    }, function () {
+                        refreshPoliciesDialog(dialogRef, domain, uuid, appliedAction, blocklist);
+                    });
+                });
             });
         }
 
-        $(document).on('click', '.block-domain', blocklist_cb);
-        $(document).on('click', '.allowlist-domain', allowlist_cb);
+        function openPoliciesDialog(domain, uuid, appliedAction, blocklist) {
+            const dialogRef = BootstrapDialog.show({
+                title: domain,
+                message: '',
+                type: BootstrapDialog.TYPE_INFO,
+                draggable: true,
+                buttons: [
+                    {
+                        label: "{{ lang._('Close') }}",
+                        action: function (d) { d.close(); }
+                    }
+                ]
+            });
 
-        do_startup().done(function() {
-            $('.wrapper').show();
-        }).fail(function() {
-            $('.wrapper').hide();
-            $('#info').show();
-        });
+            refreshPoliciesDialog(dialogRef, domain, uuid, appliedAction, blocklist);
+        }
 
         $('a[data-toggle="tab"]').on('shown.bs.tab', function (e) {
             if (e.target.id == 'query_details_tab') {
                 $("#grid-queries").bootgrid('destroy');
-                ajaxGet('/api/unbound/overview/is_block_list_enabled', {}, function(bl_enabled, status) {
-                    /* Map the command type (block/allow) to the current state of the assigned action as determined by the controller,
-                     * except for cases where they are manually overridden in the Blocklist page (Block/Allowlist Domains).
-                     */
-                    let allowlisted_domains = null;
-                    let blocklisted_domains = null;
+                ajaxGet('/api/unbound/overview/getPolicies', {}, function(policies, status) {
+                    const enabled = Object.values(policies.policies).some(v => v.enabled === "1");
                     let grid_queries = $("#grid-queries").UIBootgrid({
                         search:'/api/unbound/overview/search_queries/',
+                        commands: {
+                            action: {
+                                title: "{{ lang._('Quick action') }}",
+                                classname: 'fa fa-cogs',
+                                sequence: 100,
+                                filter: function(cell) {
```

Comment:
note to reviewer: this code utilizes https://github.com/opnsense/core/commit/d25a8acd8ed5f17c6cad005525af2245cf623ed9, which is currently only on master.